src/mkdocstrings_handlers/python/rendering.py

```diff
@@ -434,12 +436,53 @@ def do_filter_objects(
 
 
 @lru_cache(maxsize=1)
-def _get_black_formatter() -> Callable[[str, int], str]:
+def _get_formatter() -> Callable[[str, int], str]:
+    for formatter_function in [
+        _get_black_formatter,
+        _get_ruff_formatter,
+    ]:
+        if (formatter := formatter_function()) is not None:
+            return formatter
+
+    logger.info("Formatting signatures requires either Black or ruff to be installed.")
+    return lambda text, _: text
+
+
+@lru_cache(maxsize=1)
+def _get_ruff_formatter() -> Callable[[str, int], str] | None:
+    if importlib.util.find_spec("ruff") is None:
+        return None
+
+    def formatter(code: str, line_length: int) -> str:
+        try:
+            completed_process = subprocess.run(  # noqa: S603
+                [  # noqa: S607
```

Comment:
FLAG: It's non-trivial to find the correct path towards `ruff` from the current code, so I've disabled the warning here.

Reply:
Actually we can probably drop everything in the except clause and let the system handle searching in PATH. So just `return "ruff"` in the except clause. In duty I do it manually because I also search in `__pypackages__`, but we don't have to support that here.